panic in reconciler (randomsecret)

Operator is CrashLoopBackOff with following log :

```log
1.672906904078972e+09    INFO    Observed a panic in reconciler: runtime error: invalid memory address or nil pointer dereference    {"controller": "randomsecret", "controllerGroup": "redhatcop.redhat.io", "controllerKind": "RandomSecret", "RandomSecret": {"name":"my-postgresql-admin-password-geocoding","namespace":"geocoding-dev"}, "namespace": "geocoding-dev", "name": "my-postgresql-admin-password-geocoding", "reconcileID": "e575cc8a-daed-4224-a140-22201b27083b"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
    panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x1684d94]

goroutine 1001 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
    /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.0/pkg/internal/controller/controller.go:118 +0x1f4
panic({0x186b320, 0x2a566a0})
    /opt/hostedtoolcache/go/1.19.3/x64/src/runtime/panic.go:884 +0x212
github.com/redhat-cop/vault-config-operator/api/v1alpha1.(*RandomSecret).GenerateNewPassword(0xc0014ae540, {0x1d634c8?, 0xc00323c450?})
    /home/runner/work/vault-config-operator/vault-config-operator/api/v1alpha1/randomsecret_types.go:204 +0xd4
github.com/redhat-cop/vault-config-operator/api/v1alpha1.(*RandomSecret).PrepareInternalValues(...)
    /home/runner/work/vault-config-operator/vault-config-operator/api/v1alpha1/randomsecret_types.go:112
github.com/redhat-cop/vault-config-operator/controllers.(*RandomSecretReconciler).manageReconcileLogic(0xc000550ee0, {0x1d634c8, 0xc00323c450}, 0xc0014ae540)
    /home/runner/work/vault-config-operator/vault-config-operator/controllers/randomsecret_controller.go:142 +0x13e
github.com/redhat-cop/vault-config-operator/controllers.(*RandomSecretReconciler).Reconcile(0xc000550ee0, {0x1d634c8, 0xc00163fa40}, {{{0xc0021a0100?, 0x10?}, {0xc0015032f0?, 0x40f687?}}})
    /home/runner/work/vault-config-operator/vault-config-operator/controllers/randomsecret_controller.go:107 +0x68a
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x1d63420?, {0x1d634c8?, 0xc00163fa40?}, {{{0xc0021a0100?, 0x19b9b80?}, {0xc0015032f0?, 0x405094?}}})
    /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.0/pkg/internal/controller/controller.go:121 +0xc8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00032f9a0, {0x1d63420, 0xc0013f8c40}, {0x18d33c0?, 0xc002164000?})
    /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.0/pkg/internal/controller/controller.go:320 +0x33c
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00032f9a0, {0x1d63420, 0xc0013f8c40})
    /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.0/pkg/internal/controller/controller.go:273 +0x1d9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
    /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.0/pkg/internal/controller/controller.go:234 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
    /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.0/pkg/internal/controller/controller.go:230 +0x333
Stream closed EOF for vault-admin-dev/vault-config-operator-dev-5f4bf54485-j5fvc (vault-config-operator)
```

I think it happened because I removed the vault_password_policy after creating the RandomSecret, and make it fails at next refresh

--- a/api/v1alpha1/randomsecret_types.go
+++ b/api/v1alpha1/randomsecret_types.go
@@ -201,7 +201,11 @@ func (d *RandomSecret) GenerateNewPassword(context context.Context) error {
 		if err != nil {
 			return err
 		} else {
-			d.Spec.calculatedSecret = response.Data["password"].(string)
+			if response.Data != nil {
+				d.Spec.calculatedSecret = response.Data["password"].(string)
+			} else {
+				return errors.New("no data returned by password policy")
+			}
 			return nil
 		}
 	}

